src/rsb/patterns/Reader.cpp

```diff
@@ -30,55 +30,35 @@
 
 #include <rsc/threading/SynchronizedQueue.h>
 
-#include "Handler.h"
+#include "../Factory.h"
+#include "../Handler.h"
 
-#include "eventprocessing/EventReceivingStrategy.h"
-#include "eventprocessing/InRouteConfigurator.h"
-
-using namespace std;
-
-using namespace boost;
+namespace rsb {
+namespace patterns {
 
-using namespace rsb::eventprocessing;
-using namespace rsb::transport;
+Reader::Reader(const Scope&             scope,
```

Comment:
We should somehow decide on which option set and processing modes to support in the reader. In Python, the current implementation doesn't drop events and has a fixed-length queue.

Reply:
I suggest either unbounded queues or dropping events. Blocking in the queue-pushing handler doesn't seem like a good idea.

Reply:
So, let's default to unbound for the first iteration.